assets/js/components/adminbar/AdminBarImpressions.js

```diff
@@ -73,17 +76,17 @@ function AdminBarImpressions( { WidgetReportZero, WidgetReportError } ) {
 		] )
 	);
 
-	if ( ! hasFinishedResolution ) {
-		return <PreviewBlock width="auto" height="59px" />;
-	}
-
 	if ( error ) {
 		return (
 			<WidgetReportError moduleSlug="search-console" error={ error } />
 		);
 	}
 
-	if ( isZeroReport( searchConsoleData ) ) {
+	if ( ! hasFinishedResolution || isGatheringData === undefined ) {
+		return <PreviewBlock width="auto" height="59px" />;
+	}
```

Comment:
See above, why was the overall if clause moved here?

Reply:
See my comment here: https://github.com/google/site-kit-wp/pull/4326#discussion_r745451983